From 1021ba04b39b93e2decba091cf73fa65db1ab979 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 4 Apr 2019 14:32:01 +0800
Subject: [PATCH 03/30] cpufreq: imx8: pass correct argument to
 of_cpufreq_cooling_register

In upstream commit f5f263fed66f ("cpu_cooling: Make of_cpufreq_power_cooling_register()
parse DT"), the operation of parse DT has been moved to of_cpufreq_power_cooling_register.

And in upstream commit 3ebb62ffc4e4 ("cpu_cooling: Keep only one of_cpufreq*cooling_register()
helper"), of_cpufreq_power_cooling_register has been renamed to of_cpufreq_cooling_register,
so now of_cpufreq_cooling_register not need the argument "np" anymore.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/cpufreq/imx8-cpufreq.c   | 14 +-------------
 drivers/cpufreq/imx8mq-cpufreq.c | 15 +--------------
 2 files changed, 2 insertions(+), 27 deletions(-)

diff --git a/drivers/cpufreq/imx8-cpufreq.c b/drivers/cpufreq/imx8-cpufreq.c
index 5b998b5f306f..9540f06da3bf 100644
--- a/drivers/cpufreq/imx8-cpufreq.c
+++ b/drivers/cpufreq/imx8-cpufreq.c
@@ -122,21 +122,9 @@ static int imx8_cpufreq_init(struct cpufreq_policy *policy)
 
 static void imx8_cpufreq_ready(struct cpufreq_policy *policy)
 {
-	struct device_node *np = of_get_cpu_node(policy->cpu, NULL);
 	unsigned int cluster_id = topology_physical_package_id(policy->cpu);
 
-	if (of_find_property(np, "#cooling-cells", NULL)) {
-		cdev[cluster_id] = of_cpufreq_cooling_register(np, policy);
-
-		if (IS_ERR(cdev[cluster_id]) && PTR_ERR(cdev[cluster_id]) != -ENOSYS) {
-			pr_err("cpu%d is not running as cooling device: %ld\n",
-					policy->cpu, PTR_ERR(cdev[cluster_id]));
-
-			cdev[cluster_id] = NULL;
-		}
-	}
-
-	of_node_put(np);
+	cdev[cluster_id] = of_cpufreq_cooling_register(policy);
 }
 
 static struct cpufreq_driver imx8_cpufreq_driver = {
diff --git a/drivers/cpufreq/imx8mq-cpufreq.c b/drivers/cpufreq/imx8mq-cpufreq.c
index 326710dd6c17..19dfe3d6c2bc 100644
--- a/drivers/cpufreq/imx8mq-cpufreq.c
+++ b/drivers/cpufreq/imx8mq-cpufreq.c
@@ -111,20 +111,7 @@ static int imx8mq_set_target(struct cpufreq_policy *policy, unsigned int index)
 
 static void imx8mq_cpufreq_ready(struct cpufreq_policy *policy)
 {
-	struct device_node *np = of_get_cpu_node(policy->cpu, NULL);
-
-	if (of_find_property(np, "#cooling-cells", NULL)) {
-		cdev = of_cpufreq_cooling_register(np, policy);
-
-		if (IS_ERR(cdev) && PTR_ERR(cdev) != -ENOSYS) {
-			pr_err("cpu%d is not running as cooling device: %ld\n",
-					policy->cpu, PTR_ERR(cdev));
-
-			cdev = NULL;
-		}
-	}
-
-	of_node_put(np);
+	cdev = of_cpufreq_cooling_register(policy);
 }
 
 static int imx8mq_cpufreq_init(struct cpufreq_policy *policy)
-- 
2.17.1

